### PR TITLE
[BugFix] fix severals bugs of pipe

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

1. Support `ALTER PIPE SET (<property_list>)`
2. Support property `batch_files` to control max files in a batch
3. Support set task variables through property `task.<variable_name>`
4. Report error in any stages
5. Optimize performance for the scenario of millions of files
6. Modify the file_list table primary key

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 18 / 24 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|13|19|68.42%|[374, 375, 379, 380, 435, 462]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 18 / 24 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|13|19|68.42%|[374, 375, 379, 380, 435, 462]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 18 / 24 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|13|19|68.42%|[374, 375, 379, 380, 435, 462]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 81 (51.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|8|32|25.00%|[271, 299, 300, 364, 365, 366, 367, 368, 371, 372, 373, 374, 375, 390, 391, 402, 403, 429, 431, 435, 438, 447, 454, 462]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoAccessor.java|5|14|35.71%|[83, 85, 86, 88, 89, 90, 91, 93, 94]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFileRecord.java|1|2|50.00%|[95]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListTableRepo.java|14|19|73.68%|[108, 113, 114, 115, 116]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoExecutor.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipeSetProperty.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/pipe/AlterPipeSetProperty.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/pipe/AlterPipeSetProperty.java:1]: You are using the wrong license header, need to correct it as starrocks' license header.

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/pipe/AlterPipeSetProperty.java]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 111 / 159 (69.81%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFileRecord.java|1|2|50.00%|[95]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoAccessor.java|11|20|55.00%|[83, 85, 86, 88, 89, 90, 91, 93, 94]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|35|60|58.33%|[171, 298, 327, 328, 392, 393, 394, 395, 396, 399, 400, 401, 402, 403, 418, 419, 430, 431, 456, 458, 462, 465, 474, 481, 489]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|8|13|61.54%|[164, 166, 167, 168, 169]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListTableRepo.java|13|18|72.22%|[112, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|11|14|78.57%|[90, 127, 130]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeTaskDesc.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoExecutor.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipeSetProperty.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|9|9|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 111 / 159 (69.81%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFileRecord.java|1|2|50.00%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoAccessor.java|11|20|55.00%|[82, 84, 85, 87, 88, 89, 90, 92, 93]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|35|60|58.33%|[170, 297, 326, 327, 391, 392, 393, 394, 395, 398, 399, 400, 401, 402, 417, 418, 429, 430, 455, 457, 461, 464, 473, 480, 488]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|8|13|61.54%|[163, 165, 166, 167, 168]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListTableRepo.java|13|18|72.22%|[111, 116, 117, 118, 119]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|11|14|78.57%|[90, 127, 130]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeTaskDesc.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoExecutor.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipeSetProperty.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|9|9|100.00%|[]|

**Review by satanson — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Seaven — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**